C++/Greedy-Algorithm/fractional_knapsack.cpp

```diff
@@ -0,0 +1,104 @@
+
+// ====================== Problem Statement ==========================
+/*
+You are given a set of items, each with a weight and a value. 
+You are also given a knapsack with a certain capacity. 
+Your goal is to fill the knapsack with items such that the 
+total value of the selected items is maximized, but the total 
+weight of the selected items must not exceed the capacity of the knapsack. 
+You are allowed to take fractional amounts of an item, so you can take an 
+item with a weight of 3, for example, and only take 1.5 of its weight.
+
+For example, consider the following set of items:
+
+Item	Value	Weight
+1	60	10
+2	100	20
+3	120	30
+Suppose the capacity of the knapsack is 50. In this case, 
+the optimal solution would be to take the items with values 
+100 and 60 (for a total value of 160), since these two items have 
+the highest value-to-weight ratio and their total weight is less 
+than or equal to the capacity of the knapsack.
+
+The greedy approach is often used to solve the Fractional Knapsack problem. 
+The idea is to sort the items in descending order of their value-to-weight ratio, 
+and then iterate through the items, adding them to the knapsack if they fit, 
+or adding a fraction of them if they don't fit.
+*/
+
+// // ====================== Code with C++ ==========================
+#include <iostream>
+#include <algorithm>
+#include <vector>
+
+using namespace std;
+
+// Item structure to store weight and value of each item
```

Comment:
This looks more like program from this website: _https://www.geeksforgeeks.org/c-program-for-the-fractional-knapsack-problem/_

Please refer these websites instead of plagiarising the code and changing the variable name or data types of variable.

Please understand the logic and implement in you way.